pkg/encoder/encrypter.go

```diff
@@ -1,89 +1,28 @@
 package encoder
 
-import (
-	"crypto/aes"
-	"crypto/cipher"
-	"crypto/rand"
-	"crypto/sha256"
-	"encoding/base64"
-	"io"
-
-	"github.com/go-errors/errors"
-)
-
-type TokenEncrypter struct {
-	cipherMode cipher.AEAD
-}
-
-func NewTokenEncrypter(key string) (*TokenEncrypter, error) {
-	c, err := aes.NewCipher(create32ByteKey(key))
-	if err != nil {
-		return nil, err
-	}
-	gcm, err := cipher.NewGCM(c)
-	if err != nil {
-		return nil, err
-	}
-	return &TokenEncrypter{cipherMode: gcm}, nil
+type Encrypter interface {
+	Decrypt(string) ([]byte, error)
+	Encrypt([]byte) (string, error)
 }
 
-func (e TokenEncrypter) Decode(s string) ([]byte, error) {
-	if s == "" {
-		return nil, nil
-	}
-
-	decoded, err := decode(s)
-	if err != nil {
-		return nil, err
-	}
-	return decrypt(e.cipherMode, decoded)
+type NoopEncrypter struct {
+	encoder Encoder
 }
 
-func (e TokenEncrypter) Encode(data []byte) (string, error) {
-	if len(data) == 0 {
-		return "", nil
-	}
+var _ Encrypter = (*NoopEncrypter)(nil)
 
-	encrypted, err := encrypt(e.cipherMode, data)
-	if err != nil {
-		return "", err
-	}
-	return encode(encrypted), nil
+func NewNoopEncrypter(encoder Encoder) *NoopEncrypter {
```

Comment:
This comments applies here and to the `gcm_encrypter`. It feels weird to me to have the encrypter structs take an `encoder`. This breaks a strong separation of concerns.

If we want to additionally encode the encrypted value returned from the encrypter, then that should be the responsibility of the caller, not this struct. I think, compositionally, we should be passing in a TokenEncrypter and internally constructing a `base64` encoder when constructing a new OpenFGA server. For example,

```
package server

func New(dependencies *Dependencies, ...) (*Server, error) {
    ...
    tokenEncrypter  := dependencies.TokenEncrypter

    s := &Server{
        tokenEncoder: encoder.NewBase64Encoder()
        tokenEncrypter: tokenEncrypter,
        ... // other fields
    }

    return &s, nil
}
```

Reply:
Sweet, no problem! Let me know when you've done the refactor and I'll finalize my review 👍 . Thanks!

Reply:
Refactored. Thanks again. It's much better now I think. Please check it out when you have time.